### PR TITLE
pulse: adopt 'Pulse' stats library name

### PR DESCRIPTION
**Pull request by goaway**
Description: We've decided on 'Pulse' as our public name/branding for the project's client stats library and interfaces. This change adopts that naming in code for clarity.
Risk Level: Low
Testing: CI

Signed-off-by: Mike Schore <mike.schore@gmail.com>

### CODE REVIEW

**Comment by rebello95** [comment]
@goaway can you please update the docs as well: https://github.com/envoyproxy/envoy-mobile/blob/main/docs/root/api/stats.rst?

**Comment by goaway** [comment]
@rebello95 Yep, let me do that.

**Review by rebello95 — DISMISSED** [review]

File: docs/root/api/stats.rst [hunk 1]
```diff
@@ -1,29 +1,32 @@
...
+``PulseClient``
 ---------------
 
-Envoy Mobile's stats functionality currently supports two type of stats: ``Counter`` and ``Gauge``.
+Pulse is Envoy Mobile's stats library, used for capturing client application time series
+metrics. Currently two types of metrics are supported: ``Counter`` and ``Gauge``.
 
-To use Envoy Mobile's stats function, obtain an instance of ``StatsClient`` from ``Engine``
-(refer to :ref:`api_starting_envoy` for building an engine instance), and store the stats client to
+This library (like all of Enovy Mobile) is under active development.
```
Comment (junr03) [inline docs/root/api/stats.rst:13]:
```suggestion
This library (like all of Envoy Mobile) is under active development.
```

File: docs/root/api/stats.rst [hunk 2]
```diff
@@ -1,29 +1,32 @@
 .. _api_stats:
 
-Stats
+Pulse
```
Comment (rebello95) [inline docs/root/api/stats.rst:3]: Would it make sense to do `Pulse (stats library)` here?

File: library/swift/test/BUILD [hunk 3]
```diff
@@ -33,6 +47,16 @@ envoy_mobile_swift_test(
     ],
 )
 
+envoy_mobile_swift_test(
+    name = "stats_client_impl_tests",
+    srcs = [
+        "StatsClientImplTests.swift",
```
Comment (jingwei99) [inline library/swift/test/BUILD:53]: `pulse_client_impl_tests`

`PulseClientImplTests.swift`

File: docs/root/api/stats.rst (hunk 2 above)
Comment (goaway) [reply docs/root/api/stats.rst:3]: The opening description a couple lines below states it. :)

Comment (rebello95) [reply docs/root/api/stats.rst:3]: Yeah, the intent was to make it clear at a glance without making the reader scroll for context. Up to you though

**Review by rebello95 — APPROVED** [review]

**Comment by goaway** [comment]
(Both Android and iOS compiling cleanly for me locally now.)

**Review by rebello95 — APPROVED** [review]

**Pull request merged**